### DISCUSSION TITLE
docs/reference/addon-manifest/

### DISCUSSION THREAD

**Discussion by giscus** [ScratchAddons/website-v2#272] [Page comments]
# docs/reference/addon-manifest/

In order to tell the addon loader how the addon plans to work, addons use a standard addon.json file located at the root of the addon's folder.

https://scratchaddons.com/docs/reference/addon-manifest/

**Comment by SnappyRiffs** [comment]
How many tags can the addon.json have

**Reply by WorldLanguages** [reply]
There's no limit, but it makes no sense to have "website" and "editor" at the same time, for example.

**Reply by Valmontechno** [reply]
What about addons that apply on the website and editor?

**Reply by Samq64** [reply]
We generally try to avoid those.

**Comment by Onako2** [comment]
What does "versionAdded" mean exactly? Is it what version of the addon was added in ScratchAddons? Or is it like what version of ScratchAddons was when the Addon has been made? Or something else?

**Reply by GrahamSH-LLK** [reply]
@Onako2 It's what version SA was on when the addon was merged.

**Reply by Onako2** [reply]
> @Onako2 It's what version SA was on when the addon was merged.

That means I do not need to change anything? It confused me because I got [here](https://github.com/Onako2/ScratchAddons/actions/runs/5245972116/jobs/9474168299) a warning.

**Reply by CST1229** [reply]
Usually you should set it to the next version of SA (so currently `1.33.0`), updating it whenever a new version releases (or when the addon is about to be merged, but someone else will probably do that). This is used for the "New!" tag and category on the settings page.

**Reply by Onako2** [reply]
> Usually you should set it to the next version of SA (so currently `1.33.0`), updating it whenever a new version releases (or when the addon is about to be merged, but someone else will probably do that). This is used for the "New!" tag and category on the settings page.

Should i put it to `1.34.0` or `1.33.1`?

**Reply by CST1229** [reply]
`1.33.0` (as that is the upcoming version of SA, as 1.33.0 isn't out yet). Once 1.33.0 releases, you should put it to `1.34.0`, as addons usually don't get added in minor releases (changes to the 3rd number).

**Reply by Onako2** [reply]
> `1.33.0` (as that is the upcoming version of SA, as 1.33.0 isn't out yet). Once 1.33.0 releases, you should put it to `1.34.0`, as addons usually don't get added in minor releases (changes to the 3rd number).

Ok, thanks!

**Comment by iqnite** [comment]
There's a mistake in the [if](https://scratchaddons.com/docs/reference/addon-manifest/#if) section. It says that it only works in userstyles, but it also works for userscripts.

**Reply by CST1229** [reply]
You probably shouldn't use `if` for userscripts, though (due to dynamic enable/disable/settings change).

**Reply by WorldLanguages** [reply]
Using `if` with userscripts is not recommended.
